scripts/cert-manager/gateway-tls-e2e.sh

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+  cat <<'EOF'
+Usage: gateway-tls-e2e.sh [options]
+
+Automates the end-to-end walkthrough from docs/gateway-tls-validation.md:
+  1. Provision AKS prerequisites
+  2. Validate the self-signed TLS flow
+  3. Prepare Azure Key Vault assets
+  4. Transition to provided TLS and validate connectivity
+
+Options:
+      --suffix <value>         String used to derive resource names (default: current timestamp)
+      --location <region>      Azure region for the resources (default: eastus2)
+      --resource-group <name>  Azure resource group for AKS/Key Vault (default: guanzhou-<suffix>-rg)
+      --aks-name <name>        AKS cluster name (default: guanzhou-<suffix>)
+      --keyvault <name>        Azure Key Vault name (default: ddb-issuer-<suffix>)
+      --namespace <name>       Kubernetes namespace for DocumentDB (default: documentdb-preview-ns)
+      --docdb-name <name>      DocumentDB resource name (default: documentdb-preview)
+  --github-username <val>  GitHub username for operator install (optional)
+  --github-token <val>     GitHub token with read:packages scope (optional)
+      --skip-cluster           Assume AKS cluster already exists and skip creation
+      --help                  Show this message
+EOF
+}
+
+SUFFIX="$(date +%m%d%H%M)"
+LOCATION="eastus2"
+RESOURCE_GROUP=""
+AKS_NAME=""
+KEYVAULT_NAME=""
+NAMESPACE="documentdb-preview-ns"
+DOCDB_NAME="documentdb-preview"
+SKIP_CLUSTER=0
+GITHUB_USERNAME=""
+GITHUB_TOKEN=""
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --suffix)
+      SUFFIX="$2"; shift 2 ;;
+    --location)
+      LOCATION="$2"; shift 2 ;;
+    --resource-group)
+      RESOURCE_GROUP="$2"; shift 2 ;;
+    --aks-name)
+      AKS_NAME="$2"; shift 2 ;;
+    --keyvault)
+      KEYVAULT_NAME="$2"; shift 2 ;;
+    --namespace)
+      NAMESPACE="$2"; shift 2 ;;
+    --docdb-name)
+      DOCDB_NAME="$2"; shift 2 ;;
+    --github-username)
+      GITHUB_USERNAME="$2"; shift 2 ;;
+    --github-token)
+      GITHUB_TOKEN="$2"; shift 2 ;;
+    --skip-cluster)
+      SKIP_CLUSTER=1; shift ;;
+    --help|-h)
+      usage; exit 0 ;;
+    *)
+      echo "Unknown argument: $1" >&2
+      usage
+      exit 1 ;;
+  esac
+done
+
+if [[ -z "$RESOURCE_GROUP" ]]; then
+  RESOURCE_GROUP="guanzhou-${SUFFIX}-rg"
```

Comment:
We have `guanzhou` hardcoded in many places. Please replace them with param or `documentdb`